### PR TITLE
Fixes #14: remove concurrency from ConfigurationSubscription tests

### LINKED ISSUE #14: Unstable tests
```
testReloadConfiguration(com.twitter.distributedlog.config.TestConfigurationSubscription): expected:<1> but was:<null>
```
```
<failure message="expected:&lt;1&gt; but was:&lt;null&gt;" type="java.lang.AssertionError">java.lang.AssertionError: expected:&lt;1&gt; but was:&lt;null&gt;
	at org.junit.Assert.fail(Assert.java:91)
	at org.junit.Assert.failNotEquals(Assert.java:645)
	at org.junit.Assert.assertEquals(Assert.java:126)
	at org.junit.Assert.assertEquals(Assert.java:145)
	at com.twitter.distributedlog.config.TestConfigurationSubscription.testReloadConfiguration(TestConfigurationSubscription.java:76)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.FailOnTimeout$1.run(FailOnTimeout.java:28)
</failure>
```

### PR DESCRIPTION
**Pull request by mgodave**

### CODE REVIEW

File: distributedlog-core/pom.xml [hunk 1]
```diff
@@ -52,6 +52,12 @@
       <scope>test</scope>
     </dependency>
     <dependency>
+      <groupId>org.jmock</groupId>
```
Comment (mgodave) [inline distributedlog-core/pom.xml]: I realize adding a new dependency could be an issue, at least it's under the test scope.

**Comment by mgodave** [comment]
Filed #17 for failed build

File: distributedlog-core/pom.xml (hunk 1 above)
Comment (sijie) [inline distributedlog-core/pom.xml]: we love jmock. +1

**Comment by sijie** [comment]
The change looks good to me +1.

**Pull request merged**